bin/alluxio-start.sh

```diff
@@ -52,8 +52,7 @@ check_mount_mode() {
     Mount);;
     SudoMount);;
     NoMount)
-      mount | grep "${ALLUXIO_RAM_FOLDER}" > /dev/null
-      if [[ $? -ne 0 || -z ${ALLUXIO_RAM_FOLDER} ]]; then
+      if [[ $(mount | grep -c ${ALLUXIO_RAM_FOLDER}) != 0 ]]; then
```

Comment:
@peisun1115, the change on `alluxio.worker.data.folder` should work on the required RAM folder (level0). But this solution will hard code `/userXYZ` into the relative path against each storage directory (level1 and level2), and might break the path requirement on other storage layers.

Also the description of both `ALLUXIO_RAM_FOLDER` and `alluxio.worker.tieredstore.level0.dirs.path` probably need a corresponding update if the defined path should be a mount point rather than an arbitrary storage directory. I actually reuse the configuration of `ALLUXIO_RAM_FOLDER=/dev/shm/userXYZ` on many of previous versions (since v0.5.0) and never met this problem until the recent changes. Maybe the configuration and system I'm using is a special case. I'm not sure if we need to keep this backward compatibility.